runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/config/HttpBindingConfig.java

```diff
@@ -164,6 +191,241 @@ private Function<Function<String, String>, String> asAccessor(
+                if (request.headers != null)
+                {
+                    requestType.headers = new HashMap<>();
+                    for (HttpParamConfig header : request.headers)
+                    {
+                        String8FW name = new String8FW(header.name);
+                        requestType.headers.put(name, createValidator.apply(header.validator, this.resolveId));
+                    }
+                }
+                if (request.pathParams != null)
+                {
+                    requestType.pathParams = new HashMap<>();
+                    for (HttpParamConfig pathParam : request.pathParams)
+                    {
+                        requestType.pathParams.put(pathParam.name, createValidator.apply(pathParam.validator, this.resolveId));
+                    }
+                }
+                if (request.queryParams != null)
+                {
+                    requestType.queryParams = new HashMap<>();
+                    for (HttpParamConfig queryParam : request.queryParams)
+                    {
+                        requestType.queryParams.put(queryParam.name, createValidator.apply(queryParam.validator, this.resolveId));
+                    }
+                }
+                if (request.content != null)
+                {
+                    requestType.content = createValidator.apply(request.content, this.resolveId);
+                }
+                result.add(requestType);
+            }
+        }
+        return result;
+    }
+
+    public HttpRequestType resolveRequest(
+        Function<String, String> headerByName)
+    {
+        HttpRequestType result = null;
+        if (requests != null && !requests.isEmpty())
+        {
+            String path = headerByName.apply(":path");
+            String method = headerByName.apply(":method");
+            String contentType = headerByName.apply("content-type");
+            for (HttpRequestType request : requests)
+            {
+                boolean isMatch = true;
+                if (method != null && !method.equals(request.method.name()))
+                {
+                    isMatch = false;
+                }
+                if (isMatch && contentType != null && !request.contentType.contains(contentType))
+                {
+                    isMatch = false;
+                }
+                if (isMatch)
+                {
+                    isMatch = parseParams(request, path);
+                }
+                if (isMatch)
+                {
+                    result = request;
+                    break;
+                }
+            }
+        }
+        return result;
+    }
+
+    private boolean parseParams(
+        HttpRequestType request,
+        String path)
+    {
+        boolean result = false;
+        String pattern =
+            "^" +
+            request.path.replaceAll("\\{([a-zA-Z_]+)\\}", "(?<$1>.+?)") +
+            "/?(\\?(?<query0>.*))?$";
```

Comment:
Use `String.format(...)` instead of `+` concatenation.

Reply:
done